### PR TITLE
Add migration documentation

### PR DESCRIPTION
**Pull request by gustawlippa**
Added a migration guide. I'm not sure if the "Upgrade procedure" points are needed, as they double the information, but maybe the point by point guide will be clearer then enumeration of the changes.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/235457) / elasticsearch_and_cassandra_mnesia / 89683a63d1686275a8cff989e34c89e217e3cabc
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4358/235457/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4358/235457/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%402e5411be475b.2024-08-16_13.08.47/index.html)
**OK: 461** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

File: doc/migrations/6.2.1_x.x.x.md [hunk 1]
```diff
@@ -2,3 +2,94 @@
...
+
+See [Exometer configuration doc](../configuration/instrumentation.md#exometer-options) for more information.
+
+### New Prometheus endpoint
+
+Alternatively to Exometer, MongooseIM can now expose an endpoint serving Prometheus metrics.
+The Prometheus metrics provide labels, for example `host_type`.
+Because of this, the names of the metrics are usually shorter than the Exometer ones.
+
+The simplest configuration to enable it is:
```
Comment (chrzaszcz) [inline doc/migrations/6.2.1_x.x.x.md:55]: This only enables collecting metrics in MongooseIM, and doesn't enable the actual endpoint. I know it's clarified below, but maybe we could reword this?

File: doc/migrations/6.2.1_x.x.x.md [hunk 2]
```diff
@@ -2,3 +2,94 @@
 
 Hook names have been unified by removing the `_hook` prefix from the few hooks which used it,
```
Comment (chrzaszcz) [inline doc/migrations/6.2.1_x.x.x.md:3]:
```suggestion
Hook names have been unified by removing the `_hook` suffix from the few hooks which used it,
```

**Review by chrzaszcz — COMMENTED** [review]
It looks good. I think that one sentence can be improved.

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/235520) / elasticsearch_and_cassandra_mnesia / 9bfdfcaaae82d274e3fe6e7e76d602000adabd6c
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4358/235520/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4358/235520/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%401ef5ccaf8fce.2024-08-19_13.47.56/index.html)
**OK: 461** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

File: doc/migrations/6.2.1_x.x.x.md (hunk 1 above)
Comment (gustawlippa) [reply doc/migrations/6.2.1_x.x.x.md]: Right, I've changed it to include both sections as a proposed configuration.

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/235547) / elasticsearch_and_cassandra_mnesia / f200b6af5c6a0fd7736fd2cb9fc685738206d9ca
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4358/235547/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4358/235547/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%4012336e836c9f.2024-08-19_13.58.28/index.html)
**OK: 461** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Review by chrzaszcz — APPROVED** [review]
Looks good 👍

**Pull request merged**